### PR TITLE
feat: add support for Kubernetes 1.17.0-alpha.1

### PR DESCRIPTION
**Pull request by mboersma**
**Reason for Change**:
https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.17.md#changelog-since-v1160

**Issue Fixed**:

**Requirements**:
- [x] Windows artifacts uploaded to acsmirror blob store
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:
~~k8s.gcr.io/hyperkube:v1.17.0.alpha.1 ain't gonna happen, so we need to address #1842 before this can pass.~~ *Edit*: actually hyperkube was published for this release.
~~Need to publish Windows artifacts before this will pass.~~ *Edit*: Windows artifacts are published.

### CODE REVIEW

**Comment by mboersma** [comment]
A reprieve? :-)

```console
$ docker pull k8s.gcr.io/hyperkube-amd64:v1.17.0-alpha.1
v1.17.0-alpha.1: Pulling from hyperkube-amd64
Digest: sha256:2b91b0930d4b57542b3162a42b6ac94c2fba4356d749ce47d95577907447433b
Status: Image is up to date for k8s.gcr.io/hyperkube-amd64:v1.17.0-alpha.1
k8s.gcr.io/hyperkube-amd64:v1.17.0-alpha.1
```

**Comment by mboersma** [comment]
Agghh rebase hell.

File: pkg/api/k8s_versions.go [hunk 1]
```diff
@@ -48,6 +48,41 @@ var k8sComponentVersions = map[string]map[string]string{
 		"gchighthreshold":                  strconv.Itoa(DefaultKubernetesGCHighThreshold),
 		"gclowthreshold":                   strconv.Itoa(DefaultKubernetesGCLowThreshold),
 	},
+	"1.16": {
```
Comment (jackfrancis) [inline pkg/api/k8s_versions.go]: grrrrr computer-generated diffs....

File: pkg/api/k8s_versions.go [hunk 2]
```diff
@@ -48,6 +48,41 @@ var k8sComponentVersions = map[string]map[string]string{
...
+		"dockerEngine":                     "1.13.*",
+		"dashboard":                        "kubernetes-dashboard-amd64:v1.10.1",
+		"exechealthz":                      "exechealthz-amd64:1.2",
+		"addon-resizer":                    "addon-resizer:1.8.5",
+		"heapster":                         "heapster-amd64:v1.5.4",
+		"metrics-server":                   "metrics-server-amd64:v0.3.5",
```
Comment (jackfrancis) [inline pkg/api/k8s_versions.go]: Shouldn't this new metrics-server be in the 1.17 map?

Comment (mboersma) [reply pkg/api/k8s_versions.go]: Nice catch–I'll fix it.

**Pull request merged**